homeassistant/components/google_sheets/__init__.py

```diff
@@ -96,12 +99,17 @@ def _append_to_sheet(call: ServiceCall, entry: ConfigEntry) -> None:
 
     async def append_to_sheet(call: ServiceCall) -> None:
         """Append new line of data to a Google Sheets document."""
-
-        entry = cast(
-            ConfigEntry,
-            hass.config_entries.async_get_entry(call.data[DATA_CONFIG_ENTRY]),
+        entry: ConfigEntry | None = hass.config_entries.async_get_entry(
+            call.data[DATA_CONFIG_ENTRY]
         )
-        session: OAuth2Session = hass.data[DOMAIN][entry.entry_id]
+        if not entry:
+            raise HomeAssistantError(
```

Comment:
This should probably raise `ValueError` instead as the service call input from the user is bad. It's not an api call that fails or a not user caused error.

Reply:
Fixed.